### PR TITLE
Issue 24 purge folder fixed

### PR DESCRIPTION
**Pull request by kf6kjg**
Should be ready to go, though I'm pretty sure that the Cassandra method `InventoryStorage::PurgeFolderInternal()` isn't live-tested - the only place it was previously called was from `InventoryStorage::PurgeFolders()` which only seems to be exercised by a unit test: `UnitTests::TestMultiPurgeFolder()`.  There is also a call through the apparently unused method `CheckedInventoryStorage::PurgeFolders()`.

### CODE REVIEW

File: OpenSim/Region/Framework/Scenes/Scene.PacketHandlers.cs [hunk 1]
```diff
@@ -680,6 +680,56 @@ public void HandlePurgeInventoryDescendents(IClientAPI remoteClient, UUID folder
...
+                    return;
+                }
+
+                //make sure the folder is either the lost and found, the trash, or a descendant of it
+                if (!CheckFolderHeirarchyIsAppropriateForPurge(folder, userProfile))
+                {
+                    m_log.ErrorFormat(
+                        "[AGENT INVENTORY]: Not purging descendents of {0} for user {1}, folder is not part of a purgeable heirarchy",
+                        folderID, remoteClient.Name);
+                }
```
Comment (ddaeschler) [inline OpenSim/Region/Framework/Scenes/Scene.PacketHandlers.cs]: Oops.. Fall through here with no early return.. We'd complain about the hierarchy but then never break out.

Comment (kf6kjg) [inline OpenSim/Region/Framework/Scenes/Scene.PacketHandlers.cs]: Wow didn't see that!  Both HandlePurgeInventoryFolder and HandlePurgeInventoryDescendents have are missing that same early return!

Comment (ddaeschler) [inline OpenSim/Region/Framework/Scenes/Scene.PacketHandlers.cs]: Haha. Well in fairness I probably made that particular mistake, so if you patch them both up I'll be very grateful.

**Comment by kf6kjg** [comment]
Fixed!

**Comment by ddaeschler** [comment]
Looks good!

**Pull request merged**